### PR TITLE
Auto sidebar, close #17

### LINKED ISSUE #17: [Feature Request] Recursively add articles in `content/posts`
I am thinking of the possibility of dividing articles into `content/posts` into smaller separate directories, such as notifications & announcements, docs for one-among.us, docs for local community, etc. 

The official implement of this requires manually adding items in `content/.vitepress/config.ts`, while thanks to 云依 we have a `content/.vitepress/theme/utils/getPostList.ts`. 

https://github.com/one-among-us/about-site/blob/9a42e47e19ea99a3e45c0c95920a68ca8ab82ac5/content/.vitepress/theme/utils/getPostList.ts#L3-L31 

What do you think about rewriting this? Also, there must be something to update accordingly in `content/posts/index.vue`.

https://github.com/one-among-us/about-site/blob/9a42e47e19ea99a3e45c0c95920a68ca8ab82ac5/content/posts/index.vue#L21

### PR DESCRIPTION
**Pull request by BeiyanYunyi**
Close #17

### CODE REVIEW

File: content/.vitepress/config.ts [hunk 1]
```diff
@@ -26,9 +27,6 @@ export default defineConfig({
           { text: '通知公告', link: '/zh-Hans/posts' },
           { text: '联系·支持', link: '/zh-Hans/contact' },
         ],
-        sidebar: {
-          '/zh-Hans/posts': getPostList('zh-Hans'),
-        },
         footer: {
```
Comment (sauricat) [inline content/.vitepress/config.ts:51]: 感觉不错。除了一个地方可能要看下，就是 footer 都显示不出来了？

**Review by sauricat — CHANGES_REQUESTED** [review]
感觉不错。除了一个地方可能要看下，就是 footer 都显示不出来了？

**Comment by BeiyanYunyi** [comment]
According to [the document](https://vitepress.dev/reference/default-theme-footer#footer), footer will be disabled when sidebar is displayed. For example, <https://oneamongus.ca/posts> has no footer.

Maybe we can disable sidebar in some pages, or use the `docsFooter` slot to manually configure footer.

**Comment by BeiyanYunyi** [comment]
Since a2a5ec8, all home pages have footer now. The main reason is they was using global component before, instead of `home` layout.

**Review by sauricat — APPROVED** [review]

**Pull request merged**